src/OrbitGl/ThreadTrack.cpp

```diff
@@ -501,8 +506,19 @@ void ThreadTrack::UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t
         batcher->AddVerticalLine(pos, box_height, draw_data.z, color, std::move(user_data));
       }
 
-      // Use the time at boundary of the next pixel as a threshold to avoid overdraw.
-      next_pixel_start_time_ns = GetNextPixelBoundaryTimeNs(pos[0] + size[0], draw_data);
+      // Optimization when going through the scopes, as it doesn't have any sense to draw two timers
+      // in the same pixel.
+      // TODO (b/199736246): NextPixelBoundaryTimeNs should be greater than the pixel provided.
+      next_pixel_start_time_ns =
+          std::max(timer_info.end() + 1, GetNextPixelBoundaryTimeNs(pos[0] + size[0], draw_data));
+
+      node_to_draw = ordered_nodes.upper_bound(next_pixel_start_time_ns);
+
+      // The previous node should be drawn only if its ending is after next_pixel_start_time_ns.
```

Comment:
As we discussed offline: I think the purpose of `upper_bound` and the following lines is to draw timer whose time range intersects with the pixel's time range. For example,

...  |  pixel i    |  pixel i+1  |  pixel i+2  |  ...
<----------------Timer A---------------->
                                                       <---Timer B--->

In the previous code where we use `lower_bound`, we draw timer B for pixel i. 
With this change, we will draw the timer A for pixel i. 

So I think you should update the comments here.